src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -30,27 +30,16 @@ export default class MongooseAdapter {
    * by id and should return all documents.
    */
   find(type, idOrIds, fields, sorts, filters, includePaths) {
-    let model = this.getModel(this.constructor.getModelName(type));
+    const model = this.getModel(this.constructor.getModelName(type));
     let queryBuilder = new mongoose.Query(null, null, model, model.collection);
-    let pluralizer = this.inflector.plural;
-    let mode = "find", idQuery;
-    let primaryDocumentsPromise, includedResourcesPromise = Q(null);
-
-    if(idOrIds) {
-      if(typeof idOrIds === "string") {
-        mode = "findOne";
-        idQuery = idOrIds;
-      }
-      else {
-        idQuery = {"$in": idOrIds};
-      }
+    const pluralizer = this.inflector.plural;
+    let primaryDocumentsPromise;
+    let includedResourcesPromise = Q(null);
 
-      queryBuilder[mode]({"_id": idQuery});
-    }
+    let idQuery = this.constructor.makeIdQuery(idOrIds);
+    let mode = typeof idQuery === 'string' ? 'findOne' : 'find';
 
-    else {
-      queryBuilder.find();
-    }
+    queryBuilder[mode](idQuery);
```

Comment:
Well spotted, though I'm confident that won't make a difference in this case. Should I still be testing `idOrIds` just for safety's sake, or does it not matter?